pkg/kubectl/apply/strategy/merge_visitor.go

```diff
@@ -99,6 +100,9 @@ func (v mergeStrategy) doMergeMap(e map[string]apply.Element) (apply.Result, err
 		// Recursively merge the map element before adding the value to the map
 		result, err := value.Merge(v.strategic)
 		if err != nil {
+			if result.Operation == apply.CONFLICT {
```

Comment:
Why do we need this?  Can we just return the result, or even not have the CONFLICT state.